### PR TITLE
Db integrity branch

### PR DESCRIPTION
**Pull request by Csignore**
adding two more fields in donor object: telephone-extension and mobile-extension;
implement parse function for these new fields;
slight change on donor.donor-name: require non-null name field. 
#56

### CODE REVIEW

File: app/admin.py [hunk 1]
```diff
@@ -87,16 +87,16 @@ def generate_pdf(modeladmin, request, queryset):
...
         ('Details', 	     {'fields': ['want_receipt']}),
         ('Address',          {'fields': [
             'address_line', 'city', 'province', 'postal_code']})
     ]
     list_display = ('id',
                     'donor_name',
                     'email',
-                    'mobile_number',
-                    'telephone_number',
+                    'mobile_number', 'mobile_extension',
```
Comment (leesw98) [inline app/admin.py]: Please format properly

File: app/models.py [hunk 2]
```diff
@@ -23,15 +23,20 @@ class Donor(models.Model):
...
-    donor_name = models.CharField(max_length=75, verbose_name='Donor Name')
+    # no it is not now, consider the situation of extention
+    # null determine whether it complains error when parsing data; blank determine if we receive a form without name
+    donor_name = models.CharField(max_length=75, verbose_name='Donor Name', null=False, blank = True) 
     email = models.EmailField(verbose_name='E-mail')
     want_receipt = models.BooleanField(verbose_name='Tax receipt?')
     telephone_number = models.CharField(
         max_length=30, blank=True, verbose_name='Telephone #')
+    telephone_extension = models.CharField(
+        max_length=10, blank=True, verbose_name='TeleExt #')
```
Comment (leesw98) [inline app/models.py]: Perhaps 'Telephone Ext.' would be better? TeleExt probably would look a made up word to regular people.
Or another thought is to use the original telephone number but change the parser. The latter would look better in the receipt I guess

Considering: https://stackoverflow.com/a/19131360

File: app/tasks.py [hunk 3]
```diff
@@ -58,6 +58,26 @@ def addItem(donation_f, description_f, particulars_f, manufacturer_f, model_f, q
         item_bulk.append(Item(tax_receipt_no=donation_f, description=description_f, particulars=particulars_f, manufacturer=manufacturer_f, model=model_f,
                               quantity=quantity_f, working=working_f, condition=condition_f, quality=quality_f, batch=batch_f, value=value_f, verified=True))
 
+    def getTeleExt(telephone):
+        if "xxxxxx" not in telephone:
+            teleNo = telephone.partition(" ext.")[0]
```
Comment (leesw98) [inline app/tasks.py]: Can you check the different formatting in the CSV and tell me the number formatting? I think 10b csv is formatted by CSV. We probably want to actually support both 10b csv and the Weebly CSV in the long term.

File: app/tasks.py [hunk 4]
```diff
@@ -82,7 +102,7 @@ def parseDate(date_f):
     # fileObject is your csv.reader
     total_row_count = sum(1 for line in csv.reader(csvfile)) - 1
```
Comment (leesw98) [inline app/tasks.py]: We should consider using csv.DictReader later. I didn't know it was a thing before.

File: app/tasks.py [hunk 5]
```diff
@@ -111,16 +134,21 @@ def parseDate(date_f):
         value_f             = unicode(row[27], "utf-8", errors='ignore')
         customer_ref_f      = unicode(row[28], "utf-8", errors='ignore')
 
-        donor_f = getCreateDonor(donor_name_f, email_f, want_receipt_f, telephone_number_f,
-                                    mobile_number_f, address_line_f, city_f, province_f, postal_code_f, customer_ref_f)
+        telephone_number_f, telephone_extension_f = getTeleExt(telephone)
```
Comment (leesw98) [inline app/tasks.py]: :+1:

Comment (Csignore) [reply app/tasks.py]: got you! gonna fix it at the meeting tmr!

**Comment by leesw98** [comment]
As discussed, change the parser to parse the numbers and format into appropriate format of (888) 888-8888 ext. 88888 (upto 5 digit extension)

**Pull request closed without merging**